plugins/services/src/js/reducers/serviceForm/MultiContainerHealthChecks.js

```diff
@@ -3,79 +3,423 @@ import {
+    const [group, field, secondField] = path;
+
+    // ADD_ITEM does nothing more but to define an object as a value,
+    // since we cannot have more than 1 items.
+    if (type === ADD_ITEM) {
+      if (path.length !== 0) {
+        if (process.env.NODE_ENV !== 'production') {
+          throw new TypeError('Trying to ADD_ITEM on a wrong health path');
         }
-        if (`healthChecks.${index}.maxConsecutiveFailures` === joinedPath) {
-          this.healthChecks[index].maxConsecutiveFailures = parseInt(value, 10);
+
+        return newState;
+      }
+
+      return {};
+    }
+
+    // REMOVE_ITEM resets the state back to `null` since we can only
+    // have one item.
+    if (type === REMOVE_ITEM) {
+      if (path.length !== 0) {
+        if (process.env.NODE_ENV !== 'production') {
+          throw new TypeError('Trying to REMOVE_ITEM on a wrong health path');
         }
+
+        return newState;
+      }
+
+      return null;
+    }
+
+    // Format object structure according to protocol switch
+    if (group === 'protocol') {
+      switch (value) {
+        case COMMAND:
+          newState.exec = {
+            command: {}
+          };
+          delete newState.http;
+          delete newState.tcp;
+          break;
+
+        case HTTP:
+          delete newState.exec;
+          newState.http = {};
+          delete newState.tcp;
+          break;
+
+        case TCP:
+          delete newState.exec;
+          delete newState.http;
+          newState.tcp = {};
+          break;
       }
+
+      return newState;
+    }
+
+    // Assign properties
+    switch (group) {
+      case 'exec':
+        return reduceCommandHealthCheck(newState, secondField, value);
+
+      case 'http':
+        return reduceHttpHealthCheck(newState, field, value);
+
+      case 'tcp':
+        return reduceTcpHealthCheck(newState, field, value);
+
+      case 'gracePeriodSeconds':
+        newState.gracePeriodSeconds = intOrNull(value);
+        break;
+
+      case 'intervalSeconds':
+        newState.intervalSeconds = intOrNull(value);
+        break;
+
+      case 'maxConsecutiveFailures':
+        newState.maxConsecutiveFailures = intOrNull(value);
+        break;
+
+      case 'timeoutSeconds':
+        newState.timeoutSeconds = intOrNull(value);
+        break;
+
+      case 'delaySeconds':
+        newState.delaySeconds = intOrNull(value);
+        break;
     }
 
-    return this.healthChecks.map(mapHealthChecks);
+    return newState;
+  },
+
+  JSONSegmentParser(healthCheck, path) {
+    const memo = [];
+    if (ValidatorUtil.isEmpty(healthCheck)) {
+      return memo;
+    }
+
+    // Add an ADD_ITEM transaction
+    memo.push(new Transaction(path, null, ADD_ITEM));
+
+    // Parse detailed fields according to type
+    if (healthCheck.http != null) {
+      memo.push(new Transaction(path.concat(['protocol']), HTTP, SET));
+      memo.push(
+        ...parseHttpHealthCheck(healthCheck.http, path.concat(['http']), memo)
```

Comment:
To avoid creating yet another array instance. But not really sure if that's helping...